android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragmentTest.java

```diff
@@ -160,4 +163,32 @@ public void deleteUserAccessTokenInvalidTest() throws InterruptedException {
         onView(withId((R.id.navigation_settings)))
                 .check(matches(isDisplayed()));
     }
-}
+
+    @Test
+    public void logoutTest() {
+        // Navigate to settings fragment
+        onView(withId(R.id.navigation_settings))
+                .perform(click());
+
+        // Click on logout button
+        onView(withId(R.id.logout))
+                .perform(click());
+
+        // Test that clearAuthorization() works
+        assertEquals(ApplicationState.getInstance(appContext).getAccessToken(), Optional.empty());
+        assertEquals(ApplicationState.getInstance(appContext).getRefreshToken(), Optional.empty());
+        assertFalse(ApplicationState.getInstance(appContext).isAuthenticated());
+
+        // Check that email login field is displayed
+        onView(withId(R.id.login_email))
+                .check(matches(isDisplayed()));
+
+        // Check that password login field is displayed
+        onView(withId(R.id.login_password))
+                .check(matches(isDisplayed()));
+
+        // Check that login button is displayed
+        onView(withId(R.id.login_button))
+                .check(matches(isDisplayed()));
+    }
+}
```

Comment:
Here you could also add a Test if the use can navigate back to the application with pressing the backbutton. This test already exists in other Classes so it should be an hopefully easy copy and paste test